src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -132,9 +132,9 @@ public ReadOnlySequence(ReadOnlyMemory<T> readOnlyMemory)
             }
             else if (typeof(T) == typeof(char))
             {
-                // TODO: MemoryMarshal.TryGetString -- doesn't exist?
-                // https://github.com/dotnet/corefx/issues/27451
-                MemoryExtensions.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length);
+                // At this point, ReadOnlyMemory is backed by a string object, so TryGetString shouldn't fail
+                bool result = MemoryMarshal.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length);
```

Comment:
Will capturing the return value have any significant perf consequences?

Reply:
I think you are in trouble already if you are using this .ctor

Reply:
e.g. its tried to get an owned memory, then and array, now a string...

Using one of the other exact type .ctors would be better (though there isn't one for string)

Reply:
> its tried to get an owned memory, then and array, now a string...

I think we should file a separate issue to figure out how to optimize this.